pynamodb/connection/base.py

```diff
@@ -366,6 +366,7 @@ def _make_api_call(self, operation_name, operation_kwargs):
                 if is_last_attempt_for_exceptions:
                     log.debug('Reached the maximum number of retry attempts: %s', attempt_number)
                     if response:
```

Comment:
IIRC, response is falsey if it's an error:
```
>>> import requests
>>> response = requests.Response()
>>> response.status_code = 400
>>> bool(response)
False
>>> if response: print('hello')
...
```

Reply:
hmm good point let me check